lib/client.js

```diff
@@ -909,12 +909,18 @@ Request.prototype.end = function(fn){
   };
 
   // progress
+  var handleProgress = function(e){
+    if (e.total > 0) {
+      e.percent = e.loaded / e.total * 100;
+    }
+    self.emit('progress', e);
```

Comment:
I was looking to prevent the potential divide by zero error, though the result would just be `Infinity`.  Perhaps that is sufficient safety for the API.